solana/solana-ibc/programs/solana-ibc/src/storage.rs

```diff
@@ -379,6 +379,40 @@ pub(crate) struct IbcStorageInner<'a, 'b> {
     pub chain: &'a mut crate::chain::ChainData,
 }
 
+/// Struct containing message chunks
+///
+/// The struct consists of chunks of message data having the first 4 bytes
+/// indicating the total size of the message
+#[account]
+#[derive(Debug)]
+pub struct MsgChunks {
+    pub type_url: String,
+    pub value: Vec<u8>,
+}
```

Comment:
This really should be zero copy or unchecked account IMO.  Furthermore, rather than giving it a structure with type url and value, it’s better to treat it as a length-prefixed sequence of bytes.  Those can than be decoded as proto to get Any message.

Reply:
the value is length prefixed but we would still need `type_url` right?

Reply:
No. The value would be an Any proto message so it would include type url.

Reply:
Oh i see, that makes sense.